lib/phoenix_html/form.ex

```diff
@@ -391,7 +391,7 @@ defmodule Phoenix.HTML.Form do
   end
 
   defp option(group_label, group_values, extra, value)
-       when is_list(group_values) or is_map(group_values) do
+       when is_list(group_values) or (is_map(group_values) and not is_struct(group_values)) do
```

Comment:
Just slightly (micro) optimising it:

```suggestion
       when is_list(group_values) or (is_map(group_values) and not is_map_key(group_values, :__struct__)) do
```